src/SprykerEco/Zed/AmazonQuicksight/Business/Validator/QuicksightAnalyticsRequestValidator.php

```diff
@@ -0,0 +1,174 @@
+<?php
+
+/**
+ * MIT License
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerEco\Zed\AmazonQuicksight\Business\Validator;
+
+use Generated\Shared\Transfer\EnableQuicksightAnalyticsRequestTransfer;
+use Generated\Shared\Transfer\EnableQuicksightAnalyticsResponseTransfer;
+use Generated\Shared\Transfer\ErrorTransfer;
+use Generated\Shared\Transfer\QuicksightAssetBundleImportJobTransfer;
+use Generated\Shared\Transfer\QuicksightUserTransfer;
+use Generated\Shared\Transfer\ResetQuicksightAnalyticsRequestTransfer;
+use Generated\Shared\Transfer\ResetQuicksightAnalyticsResponseTransfer;
+use SprykerEco\Zed\AmazonQuicksight\AmazonQuicksightConfig;
+
+class QuicksightAnalyticsRequestValidator implements QuicksightAnalyticsRequestValidatorInterface
+{
+    /**
+     * @var string
+     */
+    protected const ERROR_MESSAGE_ENABLE_ANALYTICS_FAILED = 'Failed to enable the analytics';
+
+    /**
+     * @var string
+     */
+    protected const ERROR_MESSAGE_RESET_ANALYTICS_FAILED = 'Failed to reset the analytics';
+
+    /**
+     * @uses \SprykerEco\Zed\AmazonQuicksight\AmazonQuicksightConfig::QUICKSIGHT_USER_ROLE_AUTHOR
+     *
+     * @var string
+     */
+    protected const QUICKSIGHT_USER_ROLE_AUTHOR = 'AUTHOR';
+
+    /**
+     * @var list<string>
+     */
+    protected const QUICKSIGHT_USER_AUTHOR_ROLES = [
```

Comment:
[Extendability]: Let’s move into the module configuration in case it can be extended with one more role.”

Reply:
Resolved